### DISCUSSION TITLE
Cannot display value of a temperature sensor on a floor plan

### DISCUSSION THREAD

**Discussion by popsyann** [ExperienceLovelace/ha-floorplan#33] [Q&A] [answered]
I have been searching since many days how I could display the value of a temperature sensor in a floor plan.
I have followed many explanations but none of them is working.
What I have done:
   - insert a Text object in my svg file
   - give its 'id' the name of my temperature sensor in HA (ex sensor.living_room_temperature)
   - save the file and copy it in the folder /local/custom_ui/floorplan in my raspberry hosting Home Assistant

Then Edit the Floorplan in HA using the Raw Configuration Editor and insert the following (in bold):

  - title: Floorplan
    path: floorplan
    theme: iOS Dark Mode
    panel: true
    badges: []
    cards:
      - type: vertical-stack
        cards:
          - type: horizontal-stack
            cards:
              - type: 'custom:floorplan-card'
                full_height: true
                config:
                  image: /local/custom_ui/floorplan/crescent.svg
                  stylesheet: /local/custom_ui/floorplan/floorplan.css
                  last_motion_entity: sensor.template_last_motion
                  last_motion_class: last-motion
                  log_level: info
                  console_log_level: info
                  defaults:
                    hover_action: hover-info
                    tap_action: more-info
                  rules:
                    **- entity: sensor.living_room_temperature
                      element: sensor.living_room_temperature
                      text_template: '${entity.state ? entity.state + " °C" : "undefined"}'**

Unfortunately nothing is displayed as expected. Is there something I do wrong or could you point me to a place where I will find the answer to my problem. This has been so many days trying to make it work and failing.....

My entire floor plan is completed, and everything is working well (including fan animation, icons color selection based on sensor state values,  ). This is the only remaining hick-up. Help please.
Thanks

**Comment by pkozul** [comment, marked as answer]
Hi @popsyann

Looking at your floorplan config, it looks like some of the syntax is using the old format. There is no longer a `text_template` option. This has been replaced with the `floorplan.text_set` service.

You can see this being used in the [Ring](https://experiencelovelace.github.io/ha-floorplan/docs/example-ring/) example.

In your case, the rule should look something like this:

```
    - entity: sensor.living_room_temperature
      state_action:
        - service: floorplan.text_set
          service_data: '${entity.state ? entity.state + "°C" : "unknown"}'
```

Can you try it and let us know how it goes?

**Reply by popsyann** [reply]
@pkozul 
Thank you very much for your quick answer!
What you have suggested is working very well. Result as expected: temperature is displayed in the Living Room Area

![Screenshot_2021-01-27_14-28-13](https://user-images.githubusercontent.com/76272701/105952042-f1f56f00-60ab-11eb-8477-59934a900445.png)

Just need to tweak the font for cosmetic purpose.
Also I'll work on the .css and floorplan config to play with color depending on Temperature value.

Also I have a fundamental question: where can we find the documentation which could have allowed me to avoid using old format (instead of current one)? I think that knowing where to find information will help all users.

However, let me tell you that you have done a great job with this Integration. It's very efficient, the complete system at one glance.

**Comment by pkozul** [comment]
Hi @popsyann 

Great to hear it's working for you.

We're still working on the documentation, but for now, the 3 examples (Home, Light, Ring) we have are the best place to see how it works.

https://experiencelovelace.github.io/ha-floorplan/docs/examples/

**Reply by popsyann** [reply]
Hi @pkozul 

Would you mind telling me where the overall code for the lovelace dashboard (including the floorplan-card) is stored? 
I wish to take a backup before exploring some issues I have with my system (scheduler seems to go havoc every morning, since I have installed the latest version of Home Assistant).

Thanks in advance for your reply.

Best

**Reply by OzGav** [reply]
You can use the SNAPSHOT functionality of HA or if, like me, you are using the SAMBA SHARE addon just navigate to your homeassistant machine and then copy everything (mainly the config directory) to somewhere you want to keep a backup

**Reply by popsyann** [reply]
@OzGav 
Thank you for your reply. I already do a backup of my config folder by running a script either automatically or manually, destination being a USB dongle mounted on my Raspberry Pi. I simply wanted to know whether the Dashboard was located at a specific place.
Thanks anyway.

**Reply by OzGav** [reply]
If you have installed via HACS then probably it is in .../config/www/community

**Reply by popsyann** [reply]
@OzGav Nope. Only floorplan.js

Thanks anyway

**Reply by OzGav** [reply]
OK not sure what you are after then. The code for the floorplan layout is in .../config/ui-lovelace.yaml (edit: Unless you are using the UI editor then this wont be the case) In there you should be able to see the path to the CSS and SVG. That's about everything so if there is something else then sorry I didn't understand the question.

Edit 2: If using UI editor https://community.home-assistant.io/t/where-is-the-lovelace-config-saved/87663

**Reply by popsyann** [reply]
@OzGav Thank you very much. This is exactly what I was looking for!
Have a good day!
